app/api/auth/serializers.py

```diff
@@ -0,0 +1,141 @@
+import typing as tp
+
+from django.core.validators import validate_email
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth import get_user_model, password_validation
+
+from rest_framework import serializers
+from rest_framework_simplejwt.serializers import TokenObtainPairSerializer
+
+from api.auth.validation import validate_username_field, validate_email_field, validate_password_field
+from base import settings
+
+
+class RegisterUserSerializer(serializers.Serializer):
+    """
+    Serializer for register new user
+    """
+
+    username = serializers.CharField(max_length=100, required=True)
+    password = serializers.CharField(max_length=500, required=True)
+    repeat_password = serializers.CharField(max_length=500, required=True)
+    email = serializers.CharField(max_length=100, required=False)
+    first_name = serializers.CharField(max_length=100, required=False)
+    last_name = serializers.CharField(max_length=100, required=False)
+
+    @staticmethod
+    def validate_username(username: str) -> str:
+        return validate_username_field(username=username)
+
+    @staticmethod
+    def validate_email(email: str) -> str:
+        return validate_email_field(email=email)
+
+    @staticmethod
+    def validate_password(password: str) -> str:
+        return validate_password_field(password=password)
+
+    def validate(self, attrs: tp.Dict[str, tp.Any]) -> tp.Dict[str, tp.Any]:
+        """
+        Validate all fields
```

Comment:
it doesn't validate all fields

Reply:
i have update func description but i dont think that new description looks good